---
layout: default
title: Spatial Tile
nav_order: 1
parent: Spatial
usemathjax: true
grand_parent: UI Components
---

# Spatial Tile
{: .no_toc }

This panel was originally built to support
[CODEX](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6086938/) data and was
later expanded to include
[Visium 10x](https://www.10xgenomics.com/products/spatial-gene-expression).
These use different formats for constructing the spatial image.

<div class="code-example" markdown="1">

1. *CODEX*

    For CODEX we accept image data produced by
    [cytokit](https://github.com/hammerlab/cytokit). You will need the following:

    1. **Segmented images**

        These are `.tif` files named
        `R001_X00\*_Y00\*.tif` where the asterix
        character specifies the x and y coordinates of the tile
        (starting from 1). These images should have shape
        (z-planes, 4, height, width) where the 4 axis show:

        1. the cell segmentations filled,
        2. nucleus segmentation filled,
        3. cell segmentation outlines,
        4. nucleus segmentation outlines

        Each pixel in these segmented images
        will either have value 0, denoting no cell,
        or the value of the cell ID.

    2. **A `data.csv` file**

        This file contains information output from Cytokit. Each row
        represents a cell, and the column `rid` gives the cell index
        in the `.h5ad` file.
        The column `id` gives the cell ID for that specific tile. The column `z`
        gives which z-plane in which that particular tile has the best focus,
        and columns `'tile_x'` and `'tile_y'` give the coordinates of the tile
        for that cell.

    A folder named `images` containing the tiff files and the `data.csv` file
    need to be compressed into a `.tar.gz`
    file and uploaded to Cellar in the spatial panel.

    [Here](https://drive.google.com/file/d/1flajE4b11j6gWC03PsOuU7CUUgC45l0p/view?usp=sharing)
    is an example `tar.gz` file that corresponds to *CODEX_Florida_19-003-lymph-node-R2*
    in Cellar.

    Generating the tile will color each cell by its cluster.

    <img src="../../../images/codex-spatial-image.png" width="500" class="center"/>

    Furthermore, tiles can also be colored by protein expression

    <img src="../../../images/codex-spatial-expression.png" width="500" class="center"/>

2. *Visium 10x Spatial Transcriptomics*

    For 10X spatial data, simply upload the spatial related files
    generated by [spaceranger](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/images).
    For example, for this [Adult Mouse Brain (FFPE)](https://www.10xgenomics.com/resources/datasets/adult-mouse-brain-ffpe-1-standard-1-3-0) dataset, download the "Spatial image file"
    and upload it in the `Spatial Data` panel.

    <img src="../../../images/10x-data-page.png" width="500" class="center"/>

    There is no need to upload any spatial image data
    for the `10x_breast_cancer_w_image` dataset found on our web server.

    Note that "Feature / barcode matrix (filtered)" gene expression files
    can be uploaded to Cellar in the `Load Data` panel directly.

    <img src="../../../images/visium-spatial-tile.png" width="500" class="center"/>

    Also colored by gene expression

    <img src="../../../images/visium-expression.png" width="500" class="center"/>

</div>
